### PR TITLE
Integrate doctor

### PR DESCRIPTION
**Pull request by talatcanoguz**

### CODE REVIEW

File: lib/views/screens/addpatient_view.dart [hunk 1]
```diff
@@ -57,7 +60,13 @@ class AddPatientView extends HookWidget {
...
                 ElevatedButton(
                   onPressed: () {
-                    // Implement the logic for adding a patient here
+                    userViewModel
+                        .addPatient(emailController.text)
+                        .then((String message) {
+                      ScaffoldMessenger.of(context).showSnackBar(
+                        SnackBar(content: Text(message)),
+                      );
+                    });
```
Comment (Thomas-Chambers) [inline lib/views/screens/addpatient_view.dart:69]: unrelated to this PR I suppose but the first/last name fields are not required to add patient. Not sure if intended or should fix. I am aware that we are not using it to validate the patient, not sure if we want to check that or if this is fine (this is probably fine for demo). Also for demo we can have two phones, one for patient and one for doctor. Just thought of that.

**Comment by Thomas-Chambers** [comment]
I am just doing normal testing. Is the forward arrow on recent activity supposed to pop up with the workout details like when you click on the calendar? Does not appear that there is any code for this but I felt as such

**Comment by Thomas-Chambers** [comment]
Doctor view when I click into the text box to search it overflows and has this error in the bottom right 
image

**Comment by Thomas-Chambers** [comment]
haha I break your code
image

File: lib/services/firestore_repository.dart [hunk 2]
```diff
@@ -79,4 +79,60 @@ class FireStoreRepository {
...
+    DocumentSnapshot patientSnapshot =
+        await _firestore.collection("users").doc(patientEmail).get();
+    if (patientSnapshot.exists) {
+      //Check if the given patient is a patient
+      bool? isPatient = patientSnapshot["isPatient"];
+      if (!isPatient!) {
+        message = "Given user is not a patient.";
+      } else {
+        //Check if the doctor has any patients
+        DocumentSnapshot doctorSnapshot =
```
Comment (Thomas-Chambers) [inline lib/services/firestore_repository.dart:109]: We should check if the patient has completed signup before adding them. Otherwise we get null issues

**Comment by Thomas-Chambers** [comment]
The delete button on the doctor dashboard does not do anything

**Comment by Thomas-Chambers** [comment]
Unsure why this is happening: 
image
Stack trace is 
`I/flutter ( 4471): []
I/flutter ( 4471): []

======== Exception caught by widgets library =======================================================
The following RangeError was thrown building RecentActivity(dirty):
RangeError (index): Invalid value: Valid value range is empty: 0

The relevant error-causing widget was: 
  RecentActivity RecentActivity:file:///C:/Users/thoma/eclipse-workspace/mobile_app/lib/views/screens/doctordashboarduser_view.dart:87:26
When the exception was thrown, this was the stack: 
#0      List.[] (dart:core-patch/growable_array.dart:264:36)
#1      RecentActivity.build (package:senior_design/views/widgets/recent_activity.dart:54:40)
#2      StatelessElement.build (package:flutter/src/widgets/framework.dart:5367:49)
#3      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5297:15)
#4      Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#5      ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:5279:5)
#6      ComponentElement.mount (package:flutter/src/widgets/framework.dart:5273:5)
#7      Element.inflateWidget (package:flutter/src/widgets/framework.dart:4182:16)
#8      Element.updateChild (package:flutter/src/widgets/framework.dart:3701:20)
#9      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5322:16)
#10     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5462:11)
#11     Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#12     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2779:19)
#13     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:916:21)
#14     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:360:5)
#15     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1297:15)
#16     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1227:9)
#17     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1085:5)
#18     _invoke (dart:ui/hooks.dart:170:13)
#19     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:401:5)
#20     _drawFrame (dart:ui/hooks.dart:140:31)
====================================================================================================

======== Exception caught by widgets library =======================================================
The following StateError was thrown building RecentActivityGraphWidget(dirty, state: _RecentActivityGraphWidgetState#620b5):
Bad state: No element

The relevant error-causing widget was: 
  RecentActivityGraphWidget RecentActivityGraphWidget:file:///C:/Users/thoma/eclipse-workspace/mobile_app/lib/views/screens/doctordashboarduser_view.dart:113:26
When the exception was thrown, this was the stack: 
#0      ListBase.reduce (dart:collection/list.dart:187:22)
#1      _RecentActivityGraphWidgetState.build (package:senior_design/views/widgets/graph.dart:91:45)
#2      StatefulElement.build (package:flutter/src/widgets/framework.dart:5409:27)
#3      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5297:15)
#4      StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5462:11)
#5      Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#6      ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:5279:5)
#7      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:5453:11)`

**Comment by Thomas-Chambers** [comment]
I/flutter ( 4471): []
I/flutter ( 4471): []

======== Exception caught by widgets library =======================================================
The following RangeError was thrown building RecentActivity(dirty):
RangeError (index): Invalid value: Valid value range is empty: 0

The relevant error-causing widget was: 
  RecentActivity RecentActivity:file:///C:/Users/thoma/eclipse-workspace/mobile_app/lib/views/screens/doctordashboarduser_view.dart:87:26
When the exception was thrown, this was the stack: 
#0      List.[] (dart:core-patch/growable_array.dart:264:36)
#1      RecentActivity.build (package:senior_design/views/widgets/recent_activity.dart:54:40)
#2      StatelessElement.build (package:flutter/src/widgets/framework.dart:5367:49)
#3      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5297:15)
#4      Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#5      ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:5279:5)
#6      ComponentElement.mount (package:flutter/src/widgets/framework.dart:5273:5)
#7      Element.inflateWidget (package:flutter/src/widgets/framework.dart:4182:16)
#8      Element.updateChild (package:flutter/src/widgets/framework.dart:3701:20)
#9      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5322:16)
#10     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5462:11)
#11     Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#12     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2779:19)
#13     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:916:21)
#14     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:360:5)
#15     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1297:15)
#16     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1227:9)
#17     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1085:5)
#18     _invoke (dart:ui/hooks.dart:170:13)
#19     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:401:5)
#20     _drawFrame (dart:ui/hooks.dart:140:31)
====================================================================================================

======== Exception caught by widgets library =======================================================
The following StateError was thrown building RecentActivityGraphWidget(dirty, state: _RecentActivityGraphWidgetState#620b5):
Bad state: No element

The relevant error-causing widget was: 
  RecentActivityGraphWidget RecentActivityGraphWidget:file:///C:/Users/thoma/eclipse-workspace/mobile_app/lib/views/screens/doctordashboarduser_view.dart:113:26
When the exception was thrown, this was the stack: 
#0      ListBase.reduce (dart:collection/list.dart:187:22)
#1      _RecentActivityGraphWidgetState.build (package:senior_design/views/widgets/graph.dart:91:45)
#2      StatefulElement.build (package:flutter/src/widgets/framework.dart:5409:27)
#3      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5297:15)
#4      StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5462:11)
#5      Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#6      ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:5279:5)
#7      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:5453:11)

**Comment by Thomas-Chambers** [comment]
> ======== Exception caught by widgets library =======================================================
The following StateError was thrown building RecentActivityGraphWidget(dirty, state: _RecentActivityGraphWidgetState#b74a5):
Bad state: No element

The relevant error-causing widget was: 
  RecentActivityGraphWidget RecentActivityGraphWidget:file:///C:/Users/thoma/eclipse-workspace/mobile_app/lib/views/screens/doctordashboarduser_view.dart:113:26
When the exception was thrown, this was the stack: 
#0      ListBase.reduce (dart:collection/list.dart:187:22)
#1      _RecentActivityGraphWidgetState.build (package:senior_design/views/widgets/graph.dart:91:45)
#2      StatefulElement.build (package:flutter/src/widgets/framework.dart:5409:27)
#3      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5297:15)
#4      StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5462:11)
#5      Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#6      ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:5279:5)
#7      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:5453:11)
#8      ComponentElement.mount (package:flutter/src/widgets/framework.dart:5273:5)
#9      Element.inflateWidget (package:flutter/src/widgets/framework.dart:4182:16)
#10     Element.updateChild (package:flutter/src/widgets/framework.dart:3701:20)
#11     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5322:16)
#12     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5462:11)
#13     Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#14     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2779:19)
#15     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:916:21)
#16     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:360:5)
#17     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1297:15)
#18     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1227:9)
#19     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1085:5)
#20     _invoke (dart:ui/hooks.dart:170:13)
#21     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:401:5)
#22     _drawFrame (dart:ui/hooks.dart:140:31)
====================================================================================================

======== Exception caught by widgets library =======================================================
The following RangeError was thrown building RecentActivity(dirty):
RangeError (index): Invalid value: Valid value range is empty: 0

The relevant error-causing widget was: 
  RecentActivity RecentActivity:file:///C:/Users/thoma/eclipse-workspace/mobile_app/lib/views/screens/doctordashboarduser_view.dart:87:26
When the exception was thrown, this was the stack: 
#0      List.[] (dart:core-patch/growable_array.dart:264:36)
#1      RecentActivity.build (package:senior_design/views/widgets/recent_activity.dart:54:40)
#2      StatelessElement.build (package:flutter/src/widgets/framework.dart:5367:49)
#3      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5297:15)
#4      Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#5      ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:5279:5)
#6      ComponentElement.mount (package:flutter/src/widgets/framework.dart:5273:5)
#7      Element.inflateWidget (package:flutter/src/widgets/framework.dart:4182:16)
#8      Element.updateChild (package:flutter/src/widgets/framework.dart:3701:20)
#9      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5322:16)
#10     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5462:11)
#11     Element.rebuild (package:flutter/src/widgets/framework.dart:5016:7)
#12     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2779:19)
#13     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:916:21)
#14     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:360:5)
#15     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1297:15)
#16     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1227:9)
#17     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1085:5)
#18     _invoke (dart:ui/hooks.dart:170:13)
#19     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:401:5)
#20     _drawFrame (dart:ui/hooks.dart:140:31)

File: lib/views/screens/addpatient_view.dart (hunk 1 above)
Comment (talatcanoguz) [reply lib/views/screens/addpatient_view.dart:69]: No need to do that for demo.

File: lib/services/firestore_repository.dart (hunk 2 above)
Comment (talatcanoguz) [reply lib/services/firestore_repository.dart:109]: Good call, fixed it.

**Comment by talatcanoguz** [comment]
1. No need to create a new pop-up, all the info is on the recent activity card. I removed the button.
2. Not sure why this happens, told @nealhmakwana about it.
3. I think it is best to leave this for later because it is not needed for demo.
4. Good call, fixed this.
5. This is fixed in main.

Lmk what you think @Thomas-Chambers

**Review by Thomas-Chambers — APPROVED** [review]
lgtm!

**Pull request merged**